### PR TITLE
Add python-mumps from pypi

### PR DESCRIPTION
**Pull request by akhmerov**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by akhmerov** [comment]
@conda-forge/help-python-c I believe this is ready. Some remarks:

- I've skipped the windows build due to an upstream bug, which will hopefully be fixed soon.
- Not sure why python 3.12 isn't built.

**Comment by akhmerov** [comment]
Extra information:
- The recipe was prepared with grayskull with minimal post-processing below
- Adding binary requirements to build: numpy, mumps, python, compilers.
- Skipping windows builds and pinning library versions due to already reported bugs in mumps and metis feedstocks.

**Comment by akhmerov** [comment]
The failures are due to https://github.com/conda-forge/mumps-feedstock/issues/106; I'll fix these as soon as the upstream issue is fixed.

File: recipes/python-mumps/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+build:
+  skip: true  # [py<310]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - python
+    - cython
+    - numpy
```
Comment (xhochy) [inline recipes/python-mumps/meta.yaml:21]: These should only be in host.
```suggestion
```

File: recipes/python-mumps/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    - {{ compiler('c') }}
+  host:
+    - python
+    - cython >=3.0.2
+    - meson >=1.1.0
+    - meson-python >=0.15.0
+    - numpy >=1.24
```
Comment (xhochy) [inline recipes/python-mumps/meta.yaml:28]: This should be pinned to the oldest possible.
```suggestion
    - numpy 1.24
```

File: recipes/python-mumps/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    - mumps-seq >=5.6
+    - metis =5.1.0
+    - setuptools-scm
+    - pip
+  run:
+    - python
+    - numpy >=1.24
```
Comment (xhochy) [inline recipes/python-mumps/meta.yaml:35]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipes/python-mumps/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - scipy >=1.9
+    - mumps-seq >=5.6
+    - metis =5.1.0
+    - pytest
```
Comment (xhochy) [inline recipes/python-mumps/meta.yaml:39]: Is this really a runtime requirement?

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/python-mumps/meta.yaml (hunk 4 above)
Comment (akhmerov) [reply recipes/python-mumps/meta.yaml]: Indeed, moved to test.

File: recipes/python-mumps/meta.yaml (hunk 2 above)
Comment (akhmerov) [reply recipes/python-mumps/meta.yaml]: Resolved, and also downgraded to 1.23 to be compatible with NEP 29.

**Comment by akhmerov** [comment]
@xhochy thank you for the review! Should be good to go now.

**Review by xhochy — APPROVED** [review]

**Pull request merged**